.github/workflows/build.yml

```diff
@@ -76,10 +77,11 @@ jobs:
       - name: Matrix Variables
         run: |
           echo "BASE_IMAGE_NAME=${{ matrix.base_image_name }}" >> $GITHUB_ENV
+          echo "TARGET_CPU=${{ matrix.target_cpu }}" >> $GITHUB_ENV   
           if [[ "${{ matrix.image_flavor }}" =~ "main" ]]; then
-              echo "IMAGE_NAME=${{ format('{0}-{1}', matrix.base_image_name, matrix.base_name) }}" >> $GITHUB_ENV
+              echo "IMAGE_NAME=${{ format('{0}-{1}-{2}', matrix.base_image_name, matrix.base_name, matrix.target_cpu) }}" >> $GITHUB_ENV
           else
-              echo "IMAGE_NAME=${{ format('{0}-{1}{2}', matrix.base_image_name, matrix.base_name, matrix.image_flavor) }}" >> $GITHUB_ENV
+              echo "IMAGE_NAME=${{ format('{0}-{1}-{2}-{3}', matrix.base_image_name, matrix.base_name, matrix.target_cpu, matrix.image_flavor) }}" >> $GITHUB_ENV
```

Comment:
I was a little concerned about deviating from the naming convention here, suggestions are appreciated!

Reply:
Since the image flavor has a dash appended, it'd need to be `{0}-{1}{2}-{3}`. Howe er, deviating from the current image name could be problematic for existing users

Reply:
Good catch, I'll remove that extra dash.

Is there any other preferred way of handling this type of thing at build time? Best I can think of is a script to swap the services. Personally, I'd prefer a dedicated AMD image. Is the deviation from naming convention a show stopper?